### PR TITLE
Fix for Response status to show in On-Hold list

### PR DESCRIPTION
**Pull request by divyav-aot**

### CODE REVIEW

File: forms-flow-web/src/components/FOI/customComponents/StateDropDown.js [hunk 1]
```diff
@@ -94,37 +98,37 @@ const StateDropDown = ({
...
       _stateList = MinistryStateList;
     }
-    const personalRequest = requestType?.toLowerCase() === FOI_COMPONENT_CONSTANTS.REQUEST_TYPE_PERSONAL;
+    const personalRequest =
+      requestType?.toLowerCase() ===
+      FOI_COMPONENT_CONSTANTS.REQUEST_TYPE_PERSONAL;
     const personalIAO = !_isMinistryCoordinator && personalRequest;
-
+    const previousState =
+      stateTransition?.length > 0 && stateTransition[1]?.status;
```
Comment (divyav-aot) [inline forms-flow-web/src/components/FOI/customComponents/StateDropDown.js:106]: declared previousState in the top to check whether it is feeestimate or response. Based on that the onhold state list will be returned.

File: forms-flow-web/src/components/FOI/customComponents/StateDropDown.js [hunk 2]
```diff
@@ -136,50 +140,69 @@ const StateDropDown = ({
...
+              _state.status.toLowerCase() !==
+              StateEnum.callforrecords.name.toLowerCase()
+          );
+        } else if (previousState === StateEnum.feeassessed.name) {
+          return _stateList.onhold.filter(
+            (_state) =>
+              _state.status.toLowerCase() !==
+              StateEnum.response.name.toLowerCase()
+          );
+        }
```
Comment (divyav-aot) [inline forms-flow-web/src/components/FOI/customComponents/StateDropDown.js:175]: changed the logic of onhold state list based on previousState condition.

File: forms-flow-web/src/components/FOI/customComponents/StateDropDown.js [hunk 3]
```diff
@@ -136,50 +140,69 @@ const StateDropDown = ({
...
         return _stateList.consult;
       case StateEnum.signoff.name.toLowerCase():
         return _stateList.signoff;
       case StateEnum.feeassessed.name.toLowerCase():
-        if (personalIAO)
-          return _stateList.feeassessedforpersonal;
+        if (personalIAO) return _stateList.feeassessedforpersonal;
         return _stateList.feeassessed;
-      case StateEnum.onhold.name.toLowerCase():
-        return _stateList.onhold;
```
Comment (divyav-aot) [inline forms-flow-web/src/components/FOI/customComponents/StateDropDown.js:161]: removed unnecessary case for onhold

File: forms-flow-web/src/constants/FOI/statusEnum.js [hunk 4]
```diff
@@ -1,74 +1,214 @@
...
+  ],
+  feeassessedforpersonal: [
+    { status: "Fee Estimate", isSelected: false },
+    { status: "Call For Records", isSelected: false },
+    { status: "Closed", isSelected: false },
+  ],
+  onhold: [
+    { status: "On Hold", isSelected: false },
+    { status: "Call For Records", isSelected: false },
+    { status: "Response", isSelected: false },
```
Comment (divyav-aot) [inline forms-flow-web/src/constants/FOI/statusEnum.js:73]: added Response as part of OnHold list

**Review by richard-aot — APPROVED** [review]

**Pull request merged**